### DISCUSSION TITLE
Spring Sprites and Water Parallax

### DISCUSSION THREAD

**Discussion by Budder645** [Techokami/SonicWorldsNext#145] [Q&A]
Im probably going to show up alot, but right now Im try to change the spring sprites for the game im working on and another thing that as of now I'm trying to see if it's possible, making water in the background scroll down like it was done in Sonic 3&K in zones like Hydrocity and Launch Base.

My first issue with the springs is that I've already made custom spirtes for it, and switch them out like I've done with Sonic's sprites, but when I save them over a new slot, the custom sprites don't change, and when I've tried cycling through the different varrients of the springs in the editor (like if I switch it from facing up to facing left) It switches back to the original.

And the second question, not really an issue. But for one of the zones I want to add to this, I want to get the water in the background to scroll up and down, and give it the illusion of getting further up or down and to even pass through it and have it rise up, like how it was done in Sonic 3. I've made a background for it and as of now have just used it to get my ideas on paper to express what i want to try with it, but haven't done anything with it yet

Here is a video and background for example

https://github.com/Techokami/SonicWorldsNext/assets/159749816/018d9cd0-c9f2-4478-b8fa-f354ee24c0bb

![Aqua Lake Act 1](https://github.com/Techokami/SonicWorldsNext/assets/159749816/f481c78e-fe52-4e14-8993-bd750f036327)

**Comment by NeonSRB2** [comment]
Newer Godot 4 versions have ParalaxLayer2Ds change position I think, so what you can do is make a layer for the far back side of the water with the x-axis at 1x scrolling scale, and  one for the front using 1x scrolling scale, at first you should try putting the water texture on the back layer, and testing how it looks with the water line you can see normally. 

After that and making sure the front layer lines up with the water level, you can move the graphic to the front layer and use some math using global_position to scale it to the other layer's coordinates, like this:

```
# assuming that otherpos = the back layer's global_position, in the front layer, and h is the height...
func _process():
	var diff = otherpos - global_position
	scale.y = height / diff
```
then you can add whatever extra paralax layers as children of the one with the script, using only x scroll scale, to properly make the water texture paralax sideways.

**Comment by NeonSRB2** [comment]
Oh and the springs are just animatedSprite2ds, you can either edit the textures that already exist or import new ones in the spring scene.
